crates/viewer/re_viewer_context/src/file_dialog.rs

```diff
@@ -0,0 +1,59 @@
+use crate::ViewerContext;
+
+impl<'a> ViewerContext<'a> {
+    /// Save some bytes to disk, after first showing a save dialog.
+    #[allow(clippy::unused_self)] // Not used on Wasm
+    pub fn save_file_dialog(&self, file_name: String, title: String, data: Vec<u8>) {
```

Comment:
great to have this here. Can we also use this for rrd saving then? All this looks very suspiciously identical to `app.rs`'s `async_save_dialog`. I'd really like to keep the number of places where we have this down.

this would require a version that generates the data lazily upon save - in case the dialog is cancelled, we likely don't want to do the heavy lifting beforehand for rrd and for images that would be better as well.

Reply:
It would be nice to unify at some point, but it will require some work. I'll add a TODO